src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ExplicitConversion.cs

```diff
@@ -132,11 +132,14 @@ public bool Bind()
 
                 // But it failed, and so we know that the constant is not in range
 
+                Debug.Assert(
+                    _typeDest.GetTypeKind() != TypeKind.TK_MethodGroupType
+                    && _typeDest.GetTypeKind() != TypeKind.TK_ErrorType
+                    && _typeDest.GetTypeKind() != TypeKind.TK_ArgumentListType
+                    && _typeDest.GetTypeKind() != TypeKind.TK_ParameterModifierType
+                    && _typeDest.GetTypeKind() != TypeKind.TK_NullableType);
```

Comment:
Why is it better to list out the known bad rather than having the default case to catch anything not explicitly known good?